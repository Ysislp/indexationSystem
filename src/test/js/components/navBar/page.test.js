import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import NavBar from '../../../../main/js/components/navBar';
import Page from '../../../../main/js/components/navBar/page';

Enzyme.configure({ adapter: new Adapter() });

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('NavBar component Test', () => {

  let store;
  let getInstanceId;
  
  beforeEach(() => {

    store = mockStore({
      appInfo: {
        instanceId: 'test',
      }
    });

    getInstanceId = jest.fn();
  })

  it('Snapshot navBar', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <NavBar />
        </MemoryRouter>
      </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should show the link to /all', () => {
    const component = shallow(<Page />);
    expect(component.find('.button').at(0).exists()).toBeTruthy();
  });

  it('Should show the link to /', () => {
    const component = shallow(<Page />);
    expect(component.find('.button').at(1).exists()).toBeTruthy();
  })
})