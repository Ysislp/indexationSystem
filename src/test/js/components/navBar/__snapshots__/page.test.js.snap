// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component Test Snapshot navBar 1`] = `
<div
  className="makeStyles-root-1"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root makeStyles-menuButton-2"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.5 6v11.5c0 2.21-1.79 4-4 4s-4-1.79-4-4V5c0-1.38 1.12-2.5 2.5-2.5s2.5 1.12 2.5 2.5v10.5c0 .55-.45 1-1 1s-1-.45-1-1V6H10v9.5c0 1.38 1.12 2.5 2.5 2.5s2.5-1.12 2.5-2.5V5c0-2.21-1.79-4-4-4S7 2.79 7 5v12.5c0 3.04 2.46 5.5 5.5 5.5s5.5-2.46 5.5-5.5V6h-1.5z"
        />
      </svg>
      <h6
        className="MuiTypography-root makeStyles-title-3 MuiTypography-h6"
      >
        Indexation System  
        <sub>
          <i />
        </sub>
      </h6>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text button MuiButton-colorInherit"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <a
            className="makeStyles-links-4"
            href="/all"
            onClick={[Function]}
          >
            Ver Todos
          </a>
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text button MuiButton-colorInherit"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <a
            className="makeStyles-links-4"
            href="/"
            onClick={[Function]}
          >
            Agregar
          </a>
        </span>
      </button>
    </div>
  </header>
</div>
`;
